tasks-standard/src/main/scala/sbt/std/Streams.scala

```diff
@@ -184,6 +184,7 @@ object Streams {
       def make[T <: Closeable](a: Key, sid: String)(f: File => T): T = synchronized {
         checkOpen()
         val file = taskDirectory(a) / sid
+        Option(file.getParentFile).foreach(IO.createDirectory)
```

Comment:
I feel like touch failing is a symptom of race condition around `IO.delete` and stream used for logging than touch per se, so I don't know if this would change much.

Reply:
Could very well be. Like I said, I don't really understand what causes it or how to reproduce.

Reply:
I'll add a comment.

Reply:
ok. How about we remember to create a GitHub issue next time we see touch failure on Travis instead then?

Reply:
This has been happening relatively frequently so it probably would make sense to just go ahead and make an issue now. We haven't made any other changes that I think would mitigate it.

Reply:
Do we have a stack trace to confirm which task could trigger the error?
Ref https://github.com/sbt/sbt/pull/3093
Ref https://github.com/sbt/sbt/issues/2863